Interface

josephShokry opened the pull request:
the test will be implemented before accepting the pr

josephShokry committed:
adding the project template

josephShokry committed:
java templet

josephShokry committed:
adding the pom file

josephShokry committed:
solving mojo error

josephShokry committed:
changing java version

josephShokry committed:
Merge branch 'interface' of https://github.com/josephShokry/DB-Driver into interface

josephShokry committed:
adding the exceptions

adding all required exceptions and commenting the IDriver

josephShokry committed:
fix comments

josephShokry committed:
implementing the create

mohamedabdelsalam commented on src/main/java/main.java:
```diff
@@ -0,0 +1,15 @@
+import org.json.JSONException;
+
+import java.io.IOException;
```
You shouldn’t have a main

josephShokry replied:
ok deleted it

mohamedabdelsalam commented on .github/workflows/testcases.yml:
```diff
@@ -0,0 +1,22 @@
+name: running testCases
```
I would recommend adding in maven code convension validation. I did the same in sample calculator

josephShokry committed:
Merge branch 'main' into interface

josephShokry committed:
adding the mockito and testcases

josephShokry committed:
solving the running test

josephShokry committed:
solving the mvn errors

josephShokry committed:
fix mvn

josephShokry committed:
fix mvn

josephShokry committed:
fix mvn

josephShokry committed:
fix mvn

josephShokry committed:
fix mvn

josephShokry committed:
fix mvn

josephShokry committed:
fix mvn

josephShokry committed:
fix mvn

mohamedabdelsalam commented on src/main/java/Driver.java:
```diff
@@ -1,39 +1,57 @@
-public class Driver implements IDriver{
+import exceptions.*;
+import org.json.*;
+import java.io.*;
```
Don’t use *

josephShokry replied:
ok by why?
it reduces the number of lines
I think because it load the whole library and I don't need all of it but I think it isn't a big deal to load the whole library

mohamedabdelsalam commented on src/main/java/Driver.java:
```diff
@@ -1,39 +1,57 @@
-    public void createDatabase(String schemaPath) {
-
+    public void createDatabase(String schemaPath) throws IOException, JSONException {
+        String command = Paths.get("source","main.py") + " -cmd create -sch " + schemaPath;
```
Use string.format

josephShokry replied:
ok

josephShokry replied:
done

mohamedabdelsalam commented on src/main/java/Execute.java:
```diff
@@ -0,0 +1,27 @@
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class Execute {
```
Executor

mohamedabdelsalam commented on src/main/java/Execute.java:
```diff
@@ -0,0 +1,27 @@
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.Properties;
+
```
This class should be singlton
Read about singlton in Java

mohamedabdelsalam commented on src/test/java/operationsTest.java:
```diff
@@ -1,11 +1,41 @@
+import static org.mockito.Mockito.*;
 
 class operationsTest {
+
```
Where are the test cases for set,  get, and  delete

josephShokry replied:
In this pr i just implemented the create

josephShokry committed:
fix comments

mohamedabdelsalam commented on src/main/java/Driver.java:
```diff
@@ -1,39 +1,57 @@
-    @Override
-    public void createDatabase(String schemaPath) {
 
+public class Driver implements IDriver{
```
Space before {

mohamedabdelsalam commented on src/main/java/Driver.java:
```diff
@@ -1,39 +1,57 @@
 
+public class Driver implements IDriver{
+    private final Executer executer;
+    public Driver(final Executer executer){
```
Space before {

mohamedabdelsalam commented on src/main/java/Driver.java:
```diff
@@ -1,39 +1,57 @@
+    public Driver(final Executer executer){
+        this.executer = executer;
+    }
+    @Override
```
Add new line after }

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,35 @@
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public class Executer {
```
Final class

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,35 @@
+        FileReader confgFile = new FileReader(confgFilePath);
+        Properties confg = new Properties();
+        confg.load(confgFile);
+        databasePath = confg.getProperty("databasePath");
```
Move those to constructor and make them final

josephShokry replied:
i made those in a function and call it in the constructor to make it clear.
should I delete this function and put the code in the constructor

mohamedabdelsalam replied:
No, make this function return config 
In constructor, make database path final 
This.databasepath = config.get

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,35 @@
+        pythonCaller = confg.getProperty("pythonCaller");
+    }
+    JSONObject execute(String command) throws IOException, JSONException {
+        File dir = new File(databasePath);
```
Final

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,35 @@
+    }
+    JSONObject execute(String command) throws IOException, JSONException {
+        File dir = new File(databasePath);
+        Process process = Runtime.getRuntime().exec(pythonCaller + " " + command,null,dir);
```
Final

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,35 @@
+    JSONObject execute(String command) throws IOException, JSONException {
+        File dir = new File(databasePath);
+        Process process = Runtime.getRuntime().exec(pythonCaller + " " + command,null,dir);
+        BufferedReader outputMessage = new BufferedReader(new InputStreamReader(process.getInputStream()));
```
Final

mohamedabdelsalam commented on src/test/java/operationsTest.java:
```diff
@@ -1,11 +1,41 @@
+import exceptions.*;
```
Don’t add *

josephShokry committed:
adding the rest of the funciton

implementing the rest of the functions the set, get and delete
and adding more test cases

josephShokry committed:
fix comments

josephShokry committed:
deleting the final of executer class

mohamedabdelsalam commented on src/test/java/operationsTest.java:
```diff
@@ -1,11 +1,202 @@
+
+
+
+
```
Extra vertical spaces

josephShokry committed:
fix comments

mohamedabdelsalam commented on src/main/java/Driver.java:
```diff
@@ -1,39 +1,66 @@
+        String command = String.format("%s -cmd get -database %s -table %s -q \"%s\"",Paths.get("source","main.py").toString(),databaseName ,tableName, jsonQuery.toString());
+        JSONObject api = executer.execute(command);
+        exceptionRaiser(api);
+        String result = api.getString("result").replace("{","\"{").replace("}","}\"");
```
Why do you need this replace?

josephShokry replied:
because in the get the python return ```"result": [{josn},{json}]``` with no quotes before the ```{``` so when i pass it without the replacement the ```JSONARRAY``` throw exceptions so I should add the quotes.

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,36 @@
+import java.nio.file.Paths;
+import java.util.Properties;
+
+public final class Executer {
```
I recommend add convension checker same as Yahya and goda
I did the same in sample calculator

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,36 @@
+    private static Executer executerObject = null;
+    private Executer() throws IOException {
+        Path filePath = Paths.get(Paths.get(System.getProperty("user.dir")).toString(), "src", "main", "java", "configurations.properties");
+        loadConfg(filePath.toString());
```
Make load return config
python caller should be final along with databasePath

mohamedabdelsalam commented on src/main/java/Executer.java:
```diff
@@ -0,0 +1,36 @@
+    }
+    JSONObject execute(String command) throws IOException, JSONException {
+        final File dir = new File(databasePath);
+        final Process process = Runtime.getRuntime().exec(pythonCaller + " " + command,null,dir);
```
dbmsPath is better than databaseName

josephShokry replied:
i dont understand

josephShokry committed:
fix comments

josephShokry committed:
solving style issues

josephShokry committed:
fix line character exceed

josephShokry committed:
fix checks

josephShokry committed:
fix style

josephShokry committed:
fix
